src/messages.js

```diff
@@ -95,7 +95,7 @@ var warnings = {
   W018: "Confusing use of '{a}'.",
   W019: "Use the isNaN function to compare with NaN.",
   W020: "Read only.",
-  W021: "'{a}' is a function.",
+  W021: "'{a}' is a {b}.", // is a class or is a function
```

Comment:
I just realized that a change like that would require updating a bunch of unrelated unit tests and add a lot of noise to your patch. But it seems as though we have zero tests for reassignment of function bindings. I've opened gh-2564 to address this